### PR TITLE
Friendly `build_raw` command line interface

### PR DESCRIPTION
**Pull request by gipert**
I implemented a special keyword `orig_basename` that gets substituted in the config dictionary by the name of the original DAQ file (without extension). This makes it easier to quickly convert DAQ files on the command line. Example:
```console
$ cat orca-out-spec-cli.json
{
    "ORFlashCamADCWaveformDecoder": {
        "g{key}": {
            "key_list": [[0, 3]],
            "out_stream": "/tmp/{orig_basename}.lh5"
        }
    }
}
$ pygama build-raw -w -o orca-out-spec.json legend-testdata/data/orca/fc/L200-comm-20220519-phy-geds.orca
pygama.raw.build_raw [INFO] time elapsed: 0.04 sec
pygama.raw.build_raw [INFO] output file: /tmp/L200-comm-20220519-phy-geds.lh5 (682.289 KB)
pygama.raw.build_raw [INFO] total converted: 988.125 KB
pygama.raw.build_raw [INFO] conversion speed: 26.420 MBps
```
What do you think @jasondet? There was no way to re-use the original filename in `out_stream` otherwise.

### CODE REVIEW

File: src/pygama/cli.py [hunk 1]
```diff
@@ -71,9 +76,11 @@ def pygama_cli():
 
 def build_raw_cli(args):
     for stream in args.in_stream:
+        basename = os.path.splitext(os.path.basename(stream))[0]
         build_raw(stream, in_stream_type=args.stream_type,
                   out_spec=args.out_spec, buffer_size=args.buffer_size,
-                  n_max=args.max_rows, overwrite=args.overwrite)
+                  n_max=args.max_rows, overwrite=args.overwrite,
+                  orig_basename=basename)
```
Comment (gipert) [inline src/pygama/cli.py:83]: This is how I do it.

**Comment by jasondet** [comment]
the json for the raw buffer configuration already has a generic method of doing keyword substitutions, see the paragraph right above the sample json here:
https://legend-exp.github.io/pygama/refactor/generated/pygama.raw.html#primary-classes

Here's the python script I'm using on NERSC for generating data that makes use of this feature:
```
import sys, os, json
from pygama.raw.build_raw import build_raw
from pygama.raw.orca.orca_streamer import OrcaStreamer

if len(sys.argv) < 2:
    print(f"Usage: python {sys.argv[0]} [files]")
    print(f"e.g.: python {sys.argv[0]} /global/project/projectdirs/legend/data/lngs/l200_commissioning/orca_data/Data/p0/r2/L200_p0_r2_20220606T151738Z_f0.orca")
    sys.exit()

out_spec_json = '''
{
  "ORFlashCamADCWaveformDecoder" : {
    "ch{key:03d}/raw" : {
      "key_list" : ["*"],
      "out_stream" : "{filekey}.lh5"
    }
  }
}
'''

#for arg in sys.argv[1:]:
for arg in sys.argv[9:]:
    daq_file_path = arg

    if not os.path.exists(daq_file_path):
        print(f"File {sys.argv[1]} doesn't exist")
        sys.exit()

    filekey = os.path.basename(daq_file_path)
    dot_pos = filekey.rfind('.')
    if dot_pos >= 0: filekey = filekey[:dot_pos]
    if filekey.startswith('L200'): filekey = 'L60'+filekey[4:]

    build_raw(daq_file_path, in_stream_type='ORCA', out_spec=json.loads(out_spec_json), filekey=filekey)
```

We often can't re-use the DAQ name as-is, so something more generic is typically required than just using the input basename.

**Comment by jasondet** [comment]
I think that functionality is already available by calling `build_raw()` without giving it an `out_spec` argument, with the exception that I think in that case it writes the output to the same directory as the input files, but we can change that. So someone would just do

`$ pygama build-raw data/*.orca`

which I think is even nicer ;)

**Comment by jasondet** [comment]
I think I'm missing your argument. Can you explain in more detail? My design was:
- with no arguments, build_raw() generates an output file with the same name as the input file, with an HDF5 group for each decoder, containing a table of data for that decoder
- if users want a different output filename or HDF5 group layout, they do so by providing a json config to the out_spec argument.
- if the json config has case-specific strings, the user specifies each with a keyword of their choice, then sends in the value for each specific case in the kwargs of build_raw()

This looked to me like a codification of a specific kwarg so I thought maybe it's not necessary. But if it's providing new functionality then I'm happy to include it.

**Comment by gipert** [comment]
It is indeed a codification of a specific kwarg (with the CLI only!!!), but how do you set kwargs from the command line otherwise? We could implement something like `pygama build-raw --kwargs "var1=val1;var2=val2"`...

**Comment by jasondet** [comment]
oh I'm an idiot -- i totally missed that this was in the cli only! I thought you were baking it into build_raw() itself.

I agree this is really nice for the CLI! I'm going to merge it.

**Pull request merged**